### ISSUE TITLE
Add benchmarks

### ISSUE THREAD

**Issue by jacktuck** [hugmanrique/turbo-ws#3]
Would it be possible to add a benchmark for turbo-ws to https://github.com/hashrocket/websocket-shootout/tree/master/js ?

**Comment by hugmanrique** [comment]
I still need to fix some things before release (v1.0) but feel free to do it. I can add it myself if you prefer

**Comment by jacktuck** [comment]
@hugmanrique It's cool i'm happy to do it :). Will take a look when i get a chance.

**Comment by Codewyvern** [comment]
Would be interesting to see how it compares to 'ws' on npm.

**Comment by pocesar** [comment]
and uws on linux (on windows, it blows)

**Comment by alexhultman** [comment]
I would gladly add it to my front-page benchmark chart if you manage to make it stable enough for a basic benchmark - currently I couldn't get any numbers from it since it either crashed, stopped working or both.

**Comment by hugmanrique** [comment]
Yeah, it isn't currently working. I'm working on a fix, I forgot about a really important thing: frames don't always arrive in one buffer

**Comment by alexhultman** [comment]
I mean, you should look at passing at least Chapter 1 of Autobahn test suite before anything else - all other alternatives pass the entire suite so not being able to run even a simple benchmark is worrying given how much attention these 35 commits of yours have gotten.

Also, I'm highly skeptic of your perf. claims when it's not possible to even _test_ it in any way. You don't have any benchmarks anywhere so how can it be "blazing fast"?

**Comment by hugmanrique** [comment]
I did test it with Thor, and I'm still working on a fix, just haven't had that much time lately to work on side projects.

**Comment by alexhultman** [comment]
I don't know what I'm doing wrong but I cannot seem to get it to work whatsoever, this snippet can receive one message but not send anything. After receiving one message it completely stops working and I cannot get any kind of numbers from it:

```javascript
const Server = require('./dist/').default;

const port = 5000;
const server = new Server();

/* eslint-disable no-console */

server.listen(port).then(() => {
  console.log(`⚡ Listening on *:${port}`);
});

server.on('connection', socket => {
  socket.on('text', message => {
    socket.send(message);
  });
});
```

**Comment by alexhultman** [comment]
I was able to extract one thing: scalability numbers. My skepticism was right, it doesn't even perform better than websockets/ws.

```
ws:
Connection performance: 7.1736 connections/ms
Memory usage: 100.203 mb of user space
Memory performance: 99.7973 connections/mb


turbo-ws:
Connection performance: 2.94898 connections/ms
Memory usage: 276.969 mb of user space
Memory performance: 36.1052 connections/mb
```

It requires _more_ memory and establishes connections _slower_ than websockets/ws. Case closed.

**Comment by brechtpm** [comment]
> Case closed.

I didn't realise this was a court case, and I definitely can't remember anyone making you the judge. So please, drop the attitude and be constructive. It's only version 0.1.

**Comment by hugmanrique** [comment]
The issue you're having has already been reported on #6, and is likely the cause of the overall connection response performance degradation.

Again, I'm still working on this issue and as @brechtpm (thanks! 😀) I would like to keep this criticism constructive and would love to get suggestions on how to improve the current state of the project.

**Comment by alexhultman** [comment]
It's sad to see how the JavaScript community will, time after time, _blindly_ accept anything that claims to be "blazing fast" even without so much of a shred of proof, testing or comparison. It really is the pure opposite of how science works and it rots peoples brains from the inside. If you randomly go around claiming bullshit then be prepared to be called out for it. In Sweden we have a saying that goes something like "if you play the game, be prepared for the consequences". You can't just make up random bullshit and expect to get away with it.
